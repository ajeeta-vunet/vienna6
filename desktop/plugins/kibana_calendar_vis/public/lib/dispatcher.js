/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import d3 from 'd3';

export class Dispatcher {
  constructor() {
    this.container = null;
    this.src = null;
    this.dataTarget = null;
    this.config = null;
    this.API = null;
  }

  newMapping() {
    this.src = null;
    this.dataTarget = null;
    return this;
  }

  addAPI(API) {
    this.API = API;
    return this;
  }

  addConfig(config) {
    this.config = config;
    return this;
  }

  addContainer(container) {
    if (container instanceof HTMLElement) {
      this.container = container;
    } else {
      throw new TypeError(`invalid type ${container.toString()}, should be HTMLElement`);
    }

    return this;
  }

  addSource(src) {
    this.src = src;
    return this;
  }

  addDataTarget(dataTarget) {
    if (!/^\[data-.*\]$/.test(dataTarget)) {
      throw new TypeError(`invalid type ${typeof dataTarget}, should be in a form [data-...]`);
    }
    this.dataTarget = dataTarget;
    return this;
  }

  addEvent(event, callback) {
    if (this.source === null || this.dataTarget === null) {
      throw new TypeError(`source or dataTarget is not defined`);
    }

    const selection = d3.selectAll(this.dataTarget);
    const self = this;
    const nodes = this.src !== null ? [].slice.call(this.container.querySelectorAll(this.src)) : [];
    selection.each(function () {
      const element = d3.select(this);
      if (nodes.length !== 0 && !nodes.includes(this)) {
        return;
      }

      if (typeof callback === 'function') {
        const options = {
          API: self.API,
          config: self.config,
          container: self.container,
          source: self.src,
          dataTarget: self.dataTarget
        };
        return element.on(event, callback.bind(this, options));
      }
    });

    return this;
  }

  removeEvent(event) {
    if (this.source === null || this.dataTarget === null) {
      throw new TypeError(`source or dataTarget is not defined`);
    }

    const selection = d3.selectAll(this.dataTarget);
    selection.each(function () {
      const element = d3.select(this);
      element.on(event, null);
    });

    return this;
  }

}
