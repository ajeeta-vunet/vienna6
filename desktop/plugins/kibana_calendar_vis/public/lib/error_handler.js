/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import MarkdownIt from 'markdown-it';
import d3 from 'd3';

const markdownIt = new MarkdownIt({
  html: true,
  linkify: true
});

export const errorContainer = 'visualize-error';

export class CalendarErrorHandler {
  constructor() {
    this.el = undefined;
    this.id = undefined;
  }

  bindEl(el, id) {
    this.el = el;
    this.id = id;
    return this;
  }

  removeError() {
    d3.select(this.el)
      .attr('id', this.id)
      .selectAll(`.${errorContainer}`)
      .remove();
  }

  error(msg) {
    this.removeError();
    const div = d3.select(this.el)
      .append('div')
      .attr('id', this.id)
      // class name needs `chart` in it for the polling checkSize function
      // to continuously call render on resize
      .attr('class', `${errorContainer} chart error`);

    if (msg === 'No results found') {
      div.append('div')
        .attr('class', `text-center ${errorContainer} visualize-chart`)
        .append('div').attr('class', 'item top')
        .append('div').attr('class', 'item')
        .append('h3').html('No data to show');

      div.append('div').attr('class', 'item bottom');
    } else {
      div.append('h4').html(markdownIt.renderInline(msg));
    }
    return div;
  }
}
