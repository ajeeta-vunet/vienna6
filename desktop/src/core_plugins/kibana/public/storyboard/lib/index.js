export { saveStoryboard } from './save_storyboard';
//export { getAppStateDefaults } from './get_app_state_defaults';
