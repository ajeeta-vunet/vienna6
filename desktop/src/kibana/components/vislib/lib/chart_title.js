define(function (require) {
  return function ChartTitleFactory(d3, Private) {
    var $ = require('jquery');
    var _ = require('lodash');

    var ErrorHandler = Private(require('components/vislib/lib/_error_handler'));

    /*
     * Append chart titles to the visualization
     * arguments:
     *  el => reference to a DOM element
     */
    function ChartTitle(el) {
      if (!(this instanceof ChartTitle)) {
        return new ChartTitle(el);
      }

      this.el = el;
    }

    _(ChartTitle.prototype).extend(ErrorHandler.prototype);

    // Render chart titles
    ChartTitle.prototype.render = function () {
      d3.select(this.el).selectAll('.chart-title').call(this.draw());
      d3.select(this.el).selectAll('.chart-title').call(this.truncate());
    };

    // Return a function that truncates chart title text
    // Need to refactor this function, so that it is called inside the draw method
    ChartTitle.prototype.truncate = function () {
      return function (selection) {
        selection.each(function () {
          var div = d3.select(this);
          var dataType = this.parentNode.__data__.rows ? 'rows' : 'columns';

          var text = div.select('text');
          var textLength = text.node().getComputedTextLength();
          var maxWidth = dataType === 'rows' ? $(this).height() : $(this).width();
          var subtractionPercentage = maxWidth * 0.05;
          var str = text.text();

          // if length of text is longer than the chart div, truncate
          maxWidth = maxWidth - subtractionPercentage;
          if (textLength > maxWidth) {
            var avg = textLength / str.length;
            var end = Math.floor(maxWidth / avg);

            str = str.substr(0, end) + '...';
          }

          text.text(str);
        });
      };
    };

    // Return a callback function that appends chart titles to the visualization
    ChartTitle.prototype.draw = function () {
      var self = this;

      return function (selection) {
        selection.each(function () {
          var div = d3.select(this);
          var dataType = this.parentNode.__data__.rows ? 'rows' : 'columns';
          var width = $(this).width();
          var height = $(this).height();

          // Check if width or height are 0 or NaN
          self.validateWidthandHeight(width, height);

          div.append('svg')
            .attr('width', function () {
              if (dataType === 'rows') {
                return 15;
              }
              return width;
            })
            .attr('height', height)
            .append('text')
            .attr('transform', function () {
              if (dataType === 'rows') {
                // if `rows`, rotate the chart titles
                return 'translate(11,' + height / 2 + ')rotate(270)';
              }
              return 'translate(' + width / 2 + ',11)';
            })
            .attr('text-anchor', 'middle')
            .text(function (d) {
              return d.label;
            });
        });
      };
    };

    return ChartTitle;
  };
});
