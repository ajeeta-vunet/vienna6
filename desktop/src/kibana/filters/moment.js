define(function (require) {
	var moment = require('moment');
  var _ = require('lodash');

  require('modules')
    .get('kibana/filters')
    .filter('moment', function () {
    return function (datetime, roundUp) {
      var format = 'MMMM Do YYYY, HH:mm:ss.SSS';
      if (moment.isMoment(datetime)) return datetime.format(format);
      if (_.isDate(datetime)) return moment(datetime).format(format);
      return datetime;
    };
  });
});