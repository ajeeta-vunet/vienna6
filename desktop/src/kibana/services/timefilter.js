define(function (require) {
  var _ = require('lodash');
  var moment = require('moment');
  var datemath = require('utils/datemath');
  var module = require('modules').get('kibana');
  require('components/state_management/global_state');

  module.service('timefilter', function (Promise, globalState, $rootScope) {

    var self = this;

    // TODO: This should be disabled on route change, apps need to enable it explicitly
    var enable = false;

    // These can be date math strings or moments.
    $rootScope.time = this.time = _.defaults(globalState.time || {}, {
      from: 'now-15m',
      to: 'now'
    });

    // Only gets called if the user manually updates the URL
    globalState.onUpdate(function readFromGlobalState() {
      _.assign(self.time, globalState.time);
      castTime();
    });

    var init = function () {
      castTime();
    };

    var convertISO8601 = function (stringTime) {
      var obj = moment(stringTime, 'YYYY-MM-DDTHH:mm:ss.SSSZ', true);
      return obj.isValid() ? obj : stringTime;
    };

    var castTime = function () {
      var time = globalState.time;
      if (time && time.from) self.time.from = convertISO8601(time.from);
      if (time && time.to) self.time.to = convertISO8601(time.to);
    };

    this.enabled = function (state) {
      if (!_.isUndefined(state)) enable = state;
      return enable;
    };

    this.get = function (indexPattern) {
      var filter;
      var timefield = indexPattern.timeFieldName && _.find(indexPattern.fields, {name: indexPattern.timeFieldName});

      if (timefield) {
        var bounds = this.getBounds();
        filter = {range : {}};
        filter.range[timefield.name] = {
          gte: bounds.min,
          lte: bounds.max
        };
      }
      return filter;
    };

    this.getBounds = function (timefield) {
      return {
        min: datemath.parse(self.time.from),
        max: datemath.parse(self.time.to, true)
      };
    };

    init();

  });

});
