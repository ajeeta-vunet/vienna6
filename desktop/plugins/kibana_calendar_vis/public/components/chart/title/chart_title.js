/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import moment from 'moment';
import { GridConfig } from '../grid/grid_config';
import './chart_title.less';
import { VIS_CHART_TYPE } from '../../../lib';

export class ChartTitle extends React.Component {
  constructor(props) {
    super(props);
    const { gridConfig, chartType, dateRef, label } = this.props;
    this.gridConfig = new GridConfig(gridConfig);
    let labelToShow = label;

    if (chartType === VIS_CHART_TYPE.HEATMAP_MONTH) {
      labelToShow = moment(dateRef).format('MMM YYYY') + ': ' + labelToShow + ' per month';
    } else if (chartType === VIS_CHART_TYPE.HEATMAP_YEAR) {
      labelToShow = moment(dateRef).format('YYYY') + ': ' + labelToShow + ' per year';
    } else if (chartType === VIS_CHART_TYPE.HEATMAP_DAY) {
      labelToShow = moment(dateRef).format('MMM Do YYYY') + ': ' + labelToShow + ' per day';
    }

    this.state = {
      label: labelToShow
    };
  }

  componentDidMount() {
    this.props.renderComplete();
  }

  render() {
    const year = this.state.label.slice(0, 4);
    const [xOffset, yOffset] = this.gridConfig.get(['xOffset', 'yOffset']);

    return (
      <svg id={`title_${year}`} className="calendar-title" width="100%" height={1.25 * yOffset}>
        <text x={xOffset} y={yOffset}>
          {this.state.label}
        </text>
      </svg>
    );
  }

  componentWillUnmount() {
    this.gridConfig = null;
  }
}
