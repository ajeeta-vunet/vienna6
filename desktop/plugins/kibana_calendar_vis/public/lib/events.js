/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import $ from 'jquery';
import moment from 'moment';

export function highlight({ config, container, dataTarget }) {
  const label = this.getAttribute(dataTarget.replace(/\=".*"/, '').slice(1, -1));

  function justifyOpacity(opacity) {
    const decimalNumber = parseFloat(opacity, 10);
    const fallbackOpacity = 0.5;
    return (0 <= decimalNumber && decimalNumber <= 1) ? decimalNumber : fallbackOpacity;
  }

  const dimming = config.get('visualization:dimmingOpacity');
  $(container).find(dataTarget)
    .css('opacity', 1)
    .not((els, el) => el.getAttribute(dataTarget.slice(1, -1)) === label)
    .css('opacity', justifyOpacity(dimming));
}

export function unHighlight({ container, dataTarget }) {
  const div = $(container);
  $(dataTarget, div).css('opacity', '');
}

export function expandView({ API }) {
  const { timeFilter } = API;
  const { year, month, day } = this.dataset;
  let start;
  let end;
  if (day === undefined) {
    start = moment(new Date(year, month - 1, 1));
    end = moment(new Date(new Date(year, month, 1).getTime() - 1));
  } else {
    start = moment(new Date(year, month - 1, day, 0));
    end = moment(new Date(new Date(year, month - 1, day, 24).getTime() - 1));
  }

  timeFilter.setTime(start, end, 'absolute');
}
