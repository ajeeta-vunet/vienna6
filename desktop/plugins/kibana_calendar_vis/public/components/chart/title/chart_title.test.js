/* eslint-disable */

/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import moment from 'moment';
import { findDOMNode } from 'react-dom';
import { mount } from 'enzyme';
import { ChartTitle } from './chart_title';
import { CalendarVisConfig, VIS_CHART_TYPE } from '../../../lib/calendar_vis_config';
import { defaultParams } from '../../../default_settings';
import aggResponse from '../../../__tests__/agg_response.json';
import { calculateBounds } from 'ui/timefilter/get_time';

describe('ChartTitle - default', () => {

  let visConfig;
  let visData;
  const fakeVis = {
    params: defaultParams,
    API: {
      timeFilter: {
        getBounds: function () {
          return calculateBounds({
            from: moment(1531026000000),
            to: moment(1531026000000 + 35 * 86400000)
          });
        }
      }
    }
  };

  beforeEach(() => {
    visConfig = new CalendarVisConfig(fakeVis, defaultParams);
    visData = aggResponse.rows[0];
  });

  afterEach(() => {
    visConfig = null;
    visData = null;
  });

  it('should render a full year overview chart title', () => {
    const renderComplete = jest.fn();
    const titleWrapper = mount(
      <ChartTitle
        gridConfig={visConfig.get('grid')}
        label={visData.yAxisLabel}
        dateRef={visData.series[0].values[0].x}
        chartType={VIS_CHART_TYPE.HEATMAP_YEAR}
        renderComplete={renderComplete}
      />
    );
    const title = titleWrapper.instance();
    expect(findDOMNode(title)).toMatchSnapshot();
  });

  it('should render a full month overview chart title', () => {
    const renderComplete = jest.fn();
    const titleWrapper = mount(
      <ChartTitle
        gridConfig={visConfig.get('grid')}
        label={visData.yAxisLabel}
        dateRef={visData.series[0].values[0].x}
        chartType={VIS_CHART_TYPE.HEATMAP_MONTH}
        renderComplete={renderComplete}
      />
    );
    const title = titleWrapper.instance();
    expect(findDOMNode(title)).toMatchSnapshot();
  });

  it('shoudl render a full day overview chart title', () => {
    const renderComplete = jest.fn();
    const titleWrapper = mount(
      <ChartTitle
        gridConfig={visConfig.get('grid')}
        label={visData.yAxisLabel}
        dateRef={visData.series[0].values[0].x}
        chartType={VIS_CHART_TYPE.HEATMAP_DAY}
        renderComplete={renderComplete}
      />
    );
    const title = titleWrapper.instance();
    expect(findDOMNode(title)).toMatchSnapshot();
  });

});
