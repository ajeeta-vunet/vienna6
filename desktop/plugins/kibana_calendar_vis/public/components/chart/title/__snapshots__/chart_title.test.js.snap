// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartTitle - default shoudl render a full day overview chart title 1`] = `
<svg
  class="calendar-title"
  height="25"
  id="title_Jul "
  width="100%"
>
  <text
    x="20"
    y="20"
  >
    Jul 8th 2018: Count per day
  </text>
</svg>
`;

exports[`ChartTitle - default should render a full month overview chart title 1`] = `
<svg
  class="calendar-title"
  height="25"
  id="title_Jul "
  width="100%"
>
  <text
    x="20"
    y="20"
  >
    Jul 2018: Count per month
  </text>
</svg>
`;

exports[`ChartTitle - default should render a full year overview chart title 1`] = `
<svg
  class="calendar-title"
  height="25"
  id="title_2018"
  width="100%"
>
  <text
    x="20"
    y="20"
  >
    2018: Count per year
  </text>
</svg>
`;
